runtime/syntax/man.vim

```diff
@@ -18,6 +18,11 @@ highlight default link manOptionDesc     Constant
 highlight default link manReference      PreProc
 highlight default link manSubHeading     Function
 
+if &filetype != 'man'
+  " Loaded by the location list TOC
+  finish
```

Comment:
Yes, it's for the syntax highlighting.
https://github.com/neovim/neovim/pull/5169/files#diff-cbbabc31237d18dd55a5545b0a9858f6R34